src/query/interpreter.cpp

```diff
@@ -2275,10 +2295,23 @@ PreparedQuery PrepareCypherQuery(ParsedQuery parsed_query, std::map<std::string,
   auto hints = plan::ProvidePlanHints(&plan->plan(), plan->symbol_table());
   for (const auto &hint : hints) {
     notifications->emplace_back(SeverityLevel::INFO, NotificationCode::PLAN_HINTING, hint);
+    interpreter.LogQueryMessage(hint);
+  }
+
+  if (interpreter.IsQueryLoggingActive()) {
+    std::stringstream printed_plan;
+    plan::PrettyPrint(*dba, &plan->plan(), &printed_plan);
+    interpreter.LogQueryMessage(fmt::format("Explain plan:\n{}", printed_plan.str()));
   }
 
   TryCaching(plan->ast_storage(), frame_change_collector);
   summary->insert_or_assign("cost_estimate", plan->cost());
+  interpreter.LogQueryMessage(fmt::format("Plan cost: {}", plan->cost()));
+  bool is_profile_query = false;
+  if (interpreter.IsQueryLoggingActive()) {
+    is_profile_query = true;
+  }
```

Comment:
how does profile query depend on if query logging is active?

Reply:
is_profile_query only does the measurements, not the abort
abort is handled in the return callback from PrepareProfileQuery

Reply:
if you want, maybe you can suggest another name, since is_profile_query != query will be profiled (measured and aborted)

Reply:
resolved on the call